### ISSUE TITLE
Activity Report fails with exception

### ISSUE THREAD

**Issue by mmcleod82** [snipe/snipe-it#2480]
#### Expected Behavior (or desired behavior if a feature request)

Click on Activity Report from sidebar - runs activity report and displays.

---
#### Actual Behavior

Crashes.

ErrorException in 562e25767ca63053f871e127cc9b3d7aee72af47.php line 41:

Trying to get property of non-object (View: C:\inetpub\wwwroot\snipe-it\resources\views\reports\activity.blade.php)

---
#### Please confirm you have done the following before posting your bug report:
- [x] I have enabled debug mode
- [x] I have read [checked the Common Issues page](http://docs.snipeitapp.com/common-issues.html)

---
#### Please provide answers to these questions before posting your bug report:
- Version of Snipe-IT you're running - latest 3.3
- What OS and web server you're running Snipe-IT on - IIS8 Server 2012
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc) - manual
- If you're getting an error in your browser, include that error - ErrorException in 562e25767ca63053f871e127cc9b3d7aee72af47.php line 41:

Trying to get property of non-object (View: C:\inetpub\wwwroot\snipe-it\resources\views\reports\activity.blade.php)
- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error - Any page, clicking the 'Activity Report' from sidebar.
- If a stacktrace is provided in the error, include that too.
  First five lines:
  in 562e25767ca63053f871e127cc9b3d7aee72af47.php line 41
  2.at CompilerEngine->handleViewException(object(ErrorException), '1') in compiled.php line 15318
  3.at PhpEngine->evaluatePath('C:\inetpub\wwwroot\snipe-it\storage\framework\views/562e25767ca63053f871e127cc9b3d7aee72af47.php', array('__env' => object(Factory), 'app' => object(Application), 'MaintenanceModeEnabled' => false, 'MaintenanceModeTimestamp' => object(Carbon), 'MaintenanceModeMessage' => 'We're currently working on the site, please try again later!', 'errors' => object(ViewErrorBag), 'signedIn' => true, 'user' => object(User), 'log_actions' => object(Collection))) in compiled.php line 15353
  4.at CompilerEngine->get('C:\inetpub\wwwroot\snipe-it\resources\views/reports/activity.blade.php', array('__env' => object(Factory), 'app' => object(Application), 'MaintenanceModeEnabled' => false, 'MaintenanceModeTimestamp' => object(Carbon), 'MaintenanceModeMessage' => 'We're currently working on the site, please try again later!', 'errors' => object(ViewErrorBag), 'signedIn' => true, 'user' => object(User), 'log_actions' => object(Collection))) in compiled.php line 15185
  5.at View->getContents() in compiled.php line 15173
- Any errors that appear in your browser's error console.
- Confirm whether the error is [reproduceable on the demo](https://snipeitapp.com/demo). - no.
- Include any additional information you can find in `app/storage/logs` and your webserver's logs.
- Include what you've done so far in the installation, and if you got any error messages along the way.
- Indicate whether or not you've manually edited any data directly in the database

**Comment by snipe** [comment]
> Indicate whether or not you've manually edited any data directly in the database

Also please tell us what's on line 30-50 in 562e25767ca63053f871e127cc9b3d7aee72af47.php.

**Comment by mmcleod82** [comment]
A long while ago, I did once and only once, run a query on the database to strip the first letter of a bunch of serial numbers, as they had been imported wrong. That's the only manual edits on the db.

Ok - so this is line 27-53 in 562e25767ca63053f871e127cc9b3d7aee72af47.php:

```
<thead>
                <tr role="row">
                <th class="col-sm-1"><?php echo e(trans('general.admin')); ?></th>
                <th class="col-sm-1"><span class="line"></span><?php echo e(trans('general.action')); ?></th>
                <th class="col-sm-1"><span class="line"></span><?php echo e(trans('general.type')); ?></th>
                <th class="col-sm-1"><span class="line"></span><?php echo e(trans('general.item')); ?></th>
                <th class="col-sm-1"><span class="line"></span><?php echo e(trans('general.user')); ?></th>
                <th class="col-sm-1"><span class="line"></span><?php echo e(trans('general.date')); ?></th
            </tr>
        </thead>
        <tbody>

            <?php foreach($log_actions as $log_action): ?>
            <tr>
                <td><a href="../admin/users/<?php echo e($log_action->adminlog->id); ?>/view"><?php echo e($log_action->adminlog->fullName()); ?></a></td>
                <td><?php echo e($log_action->action_type); ?></td>
                <td>
    	            <?php if($log_action->asset_type=="hardware"): ?>
    	            	Asset
    	            <?php elseif($log_action->asset_type=="software"): ?>
    	            	License
    	            <?php elseif($log_action->asset_type=="accessory"): ?>
    	            	Accessory
                    <?php elseif($log_action->asset_type=="consumable"): ?>
        	            Consumable
    	            <?php endif; ?>
                </td>
```

**Comment by snipe** [comment]
So `<?php if($log_action->asset_type=="hardware"): ?>` is line 41?

**Comment by mmcleod82** [comment]
No line 41 is `<td><a href="../admin/users/<?php echo e($log_action->adminlog->id); ?>/view"><?php echo e($log_action->adminlog->fullName()); ?></a></td>`

**Comment by snipe** [comment]
Do you have any deleted admin users?

**Comment by mmcleod82** [comment]
Can't recall - quite possible I've deleted an admin user in the early days, but it would only of been through the interface, not directly in DB 😄

**Comment by snipe** [comment]
Would you perhaps have purged deleted items at some point? That may have removed the admin from the DB completely.

**Comment by mmcleod82** [comment]
I purge deleted items regularly, in my mind to keep things tidy!

So it's looking to display history for a user that no longer exists?

**Comment by snipe** [comment]
I think so? I'll run some tests and if I can repro, I'll patch it immediately. But that definitely sounds like what's happening, and if it is, that's on us.

**Comment by mmcleod82** [comment]
snipe - this is fixed on latest 3.5 release. Working flawlessly cheers.
